pinot-core/src/main/java/org/apache/pinot/core/plan/maker/InstancePlanMakerImplV2.java

```diff
@@ -235,6 +236,10 @@ private void applyQueryOptions(QueryContext queryContext) {
 
   @Override
   public PlanNode makeSegmentPlanNode(IndexSegment indexSegment, QueryContext queryContext) {
+    ExpressionTypeResolver validator =
+        new ExpressionTypeResolver(indexSegment, queryContext);
+    queryContext.getSelectExpressions().forEach(ec -> ec.visit(validator));
```

Comment:
Review Note: I didn't spend much time figuring out where to plug this in, this is definitely not an ideal place - just the easiest to prove that the approach works